### PR TITLE
Annotate the intensification module

### PR DESCRIPTION
**Pull request by mfeurer**

### CODE REVIEW

File: smac/intensification/successive_halving.py [hunk 1]
```diff
@@ -19,6 +19,10 @@
 
 
 class SuccessiveHalving(AbstractRacer):
+    curr_inst_idx = 0
+    running_challenger = None
+    curr_challengers = set()  # type: typing.Set[Configuration]
```
Comment (ashraaghav) [inline smac/intensification/successive_halving.py]: Why define only 3 of the stage variables outside init?

Comment (mfeurer) [reply smac/intensification/successive_halving.py]: `mypy` is able to infer the types of the other variables. For a `set()` it's unclear what will be in there so I had to define it.

Comment (ashraaghav) [reply smac/intensification/successive_halving.py]: Yes, but I was wondering why the three variables here (`curr_inst_idx`, `running_challenger`, `curr_challengers`) are initialized outside of `__init__` as global class variables.

File: smac/intensification/intensification.py [hunk 2]
```diff
@@ -461,6 +466,7 @@ def _get_instances_to_run(self,
 
         # Line 11
         self.rs.shuffle(missing_runs)
+        assert N > 0
```
[the remark this answers is not in the document]
Comment (mfeurer) [reply smac/intensification/intensification.py]: Yes, will change.

File: smac/intensification/successive_halving.py (hunk 1 above)
Comment (mfeurer) [reply smac/intensification/successive_halving.py]: It seems that this is not necessary and the variables can be defined in the init, too. Will change.

**Pull request merged**